telegram/tests/test_telegram.py

```diff
@@ -0,0 +1,201 @@
+        for req in self.pending_requests:
+            if not req.finished:
+                yield req.finish()
+
+    def test_translate_inbound_message_from_channel(self):
+        default_channel = {
+            'id': 'Default channel',
+            'type': 'channel',
+        }
+        inbound_msg = {
+            'message_id': 'Message from Telegram channel',
+            'chat': default_channel,
+            'text': 'Hi from Telegram channel!',
+        }
+
+        message = self.transport.translate_inbound_message(inbound_msg)
+        self.assertEqual(inbound_msg['text'], message['content'])
+        self.assertEqual(self.bot_username, message['to_addr'])
+        self.assertEqual(default_channel['id'], message['from_addr'])
+
+    def test_translate_inbound_message_from_user(self):
+        inbound_msg = {
+            'message_id': 'Message from Telegram user',
+            'chat': 'Random chat',
+            'text': 'Hi from Telegram user!',
+            'from': self.default_user,
+        }
+
+        message = self.transport.translate_inbound_message(inbound_msg)
+        self.assertEqual(inbound_msg['text'], message['content'])
+        self.assertEqual(self.bot_username, message['to_addr'])
+        self.assertEqual(self.default_user['id'], message['from_addr'])
+
+    def test_translate_inbound_message_no_text(self):
+        inbound_msg = {
+            'message_id': 'Message without text',
+            'chat': 'Random chat',
+            'from': self.default_user,
+        }
+
+        message = self.transport.translate_inbound_message(inbound_msg)
+        self.assertEqual('', message['content'])
+        self.assertEqual(self.bot_username, message['to_addr'])
+        self.assertEqual(self.default_user['id'], message['from_addr'])
+
+    @inlineCallbacks
+    def test_inbound_update(self):
+        default_update = json.dumps({
+            'update_id': 'update_id',
+            'message': {
+                'message_id': 'msg_id',
+                'from': self.default_user,
+                'chat': {
+                    'id': 'chat_id',
+                    'type': self.PRIVATE
+                },
+                'date': 1234,
+                'text': 'Incoming message from Telegram!',
+            }
+        })
+        res = yield self.helper.mk_request(_method='POST',
+                                           _data=default_update)
+        self.assertEqual(res.code, 200)
+
+        [msg] = yield self.helper.wait_for_dispatched_inbound(1)
+
+        expected_update = json.loads(default_update)
+        self.assertEqual(msg['to_addr'], self.bot_username)
+        self.assertEqual(msg['from_addr'], self.default_user['id'])
+        self.assertEqual(msg['content'], expected_update['message']['text'])
+        self.assertEqual(msg['transport_type'],
+                         self.transport.transport_type)
+        self.assertEqual(msg['transport_name'],
+                         self.transport.transport_name)
```

Comment:
Could we add a test for:
  - when we get non-text messages
  - when we get non-message data from telegram